### PR TITLE
Auto detect labels

### PR DESCRIPTION
**Pull request by KirillPamPam**
Detect node client type and  whether a node is archive or not

### CODE REVIEW

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/calls/DefaultEthereumMethods.kt [hunk 1]
```diff
@@ -173,6 +173,10 @@ class DefaultEthereumMethods(
 
     private fun getChainSpecificMethods(chain: Chain): List<String> {
         return when (chain) {
+            Chain.ETHEREUM__GOERLI, Chain.ETHEREUM__MAINNET -> listOf(
```
Comment (Termina1) [inline src/main/kotlin/io/emeraldpay/dshackle/upstream/calls/DefaultEthereumMethods.kt:176]: SEPOLIA also

Comment (a10zn8) [inline src/main/kotlin/io/emeraldpay/dshackle/upstream/calls/DefaultEthereumMethods.kt:176]: I think we need to add all ETHEREUM__* networks , or maybe it relevant to all evm based networks?

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/ethereum/subscribe/EthereumLabelsDetector.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    private fun detectArchiveNode(): Mono<Pair<String, String>> {
+        return reader
+            .read(JsonRpcRequest("eth_getBalance", listOf("0x756F45E3FA69347A9A973A725E3C98bC4db0b5a0", "0x1")))
+            .flatMap(JsonRpcResponse::requireResult)
+            .map { "archive" to "true" }
+            .onErrorResume { Mono.empty() }
+    }
+
+    private fun nodeType(nodeType: String): String? {
```
Comment (a10zn8) [inline src/main/kotlin/io/emeraldpay/dshackle/upstream/ethereum/subscribe/EthereumLabelsDetector.kt:48]: add bor please, like bor/v0.4.0/linux-amd64/go1.19.10

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/ethereum/EthereumLikeRpcUpstream.kt [hunk 3]
```diff
@@ -67,6 +69,12 @@ open class EthereumLikeRpcUpstream(
             validatorSubscription = validator.start()
                 .subscribe(this::setStatus)
         }
+        labelsDetector.detectLabels()
```
Comment (a10zn8) [inline src/main/kotlin/io/emeraldpay/dshackle/upstream/ethereum/EthereumLikeRpcUpstream.kt:72]: is it blocks startup or not?

**Review by Termina1 — DISMISSED** [review]

**Review by a10zn8 — APPROVED** [review]

**Pull request merged**